laws/src/main/scala/cats/laws/discipline/MiniFloat.scala

```diff
@@ -0,0 +1,177 @@
+      extends MiniFloat(significand * math.pow(Finite.base.toDouble, exponent.toDouble).toFloat)
+
+  private[MiniFloat] object Finite {
+
+    private[MiniFloat] val base = 2
+
+    private val minSignificand = -2
+    private val maxSignificand = 2
+
+    private val minExponent = -1
+    private val maxExponent = 2
+
+    val allValues: List[Finite] = {
+      for {
+        significand <- Range.inclusive(minSignificand, maxSignificand)
+        exponent <- Range.inclusive(minExponent, maxExponent)
+      } yield new Finite(significand, exponent)
+    }.toList.ordDistinct(Order.by[Finite, Float](_.toFloat)).sortBy(_.toFloat)
+
+    private[MiniFloat] val allValuesAsFloats: List[Float] = allValues.map(_.toFloat)
+
+    val zero = new Finite(0, 0)
+    val max = new Finite(maxSignificand, maxExponent)
+    val min = new Finite(minSignificand, maxExponent)
+    val minPositive = new Finite(significand = 1, exponent = minExponent)
+
+    /**
+     * Returns `None` if the given float cannot fit in an instance of `Finite`.
+     */
+    def from(float: Float): Option[Finite] = {
+      val exponent: Int = getExponent(float)
+      val significand: Int = math.round(float / math.pow(Finite.base.toDouble, exponent.toDouble).toFloat)
+
+      if (significand == 0 || exponent < minExponent) {
+        Some(zero)
+      } else if (withinBounds(significand, exponent)) {
+        Some(new Finite(significand, exponent))
+      } else if (exponent > maxExponent) {
+        try {
+          val ordersOfMagnitudeToShift = Math.subtractExact(exponent, maxExponent)
+
+          val proposedSignificand: Int = Math.multiplyExact(
+            significand,
+            math.pow(base.toDouble, ordersOfMagnitudeToShift.toDouble).toInt
+          )
+          val proposedExponent: Int = Math.subtractExact(exponent, ordersOfMagnitudeToShift)
+
+          if (withinBounds(proposedSignificand, proposedExponent)) {
+            Some(new Finite(proposedSignificand, proposedExponent))
+          } else {
+            None
+          }
+        } catch {
+          case _: ArithmeticException => None
+        }
+      } else {
+        None
+      }
+    }
+
+    private def withinBounds(significand: Int, exponent: Int): Boolean =
+      (minExponent <= exponent && exponent <= maxExponent) &&
+        (minSignificand <= significand && significand <= maxSignificand)
+
+    private val floatExponentStartBit: Int = 23
+    private val floatExponentLength: Int = 8
+    private val floatExponentBias: Int = 127
+    private val floatExponentMask: Int = ((1 << floatExponentLength) - 1) << floatExponentStartBit
+
+    // This does the same thing as java.lang.Math.getExponent, but that method is not available in scalaJS so we have to
+    // do the same thing here.
+    private def getExponent(float: Float): Int =
+      ((floatExponentMask & java.lang.Float.floatToIntBits(float)) >> floatExponentStartBit) - floatExponentBias
```

Comment:
Note that I've had to write a dedicated function to extract the exponent from a `Float` since `java.lang.Math.getExponent` is not currently defined in ScalaJS. I'll probably look to add this over there soon.